tmt/steps/provision/__init__.py

```diff
@@ -429,22 +429,32 @@ def _export_environment(self, execute_environment=None):
         # override environment provided to execute().
         environment.update(self.parent.plan.environment)
         # Prepend with export and run as a separate command.
-        if not environment:
-            return ''
-        return 'export {}; '.format(
-            ' '.join(tmt.utils.shell_variables(environment)))
+        return {} if not environment else environment
+
+    @staticmethod
+    def _export_environment_str(environment):
+        """ Prepare shell export of environment variables """
+        return f'export {" ".join(tmt.utils.shell_variables(environment))}; ' \
+            if environment \
+            else ''
 
     def ansible(self, playbook, extra_args=None):
         """ Prepare guest using ansible playbook """
         playbook = self._ansible_playbook_path(playbook)
+        self.run(f'stty cols {tmt.utils.OUTPUT_WIDTH};:',
+                 shell=True)
+        ssh_common_args = [f'--ssh-common-args="{i}"'
+                           for i in self._ssh_options(join=True).split()]
         stdout, stderr = self.run(
-            f'{self._export_environment()}'
-            f'stty cols {tmt.utils.OUTPUT_WIDTH}; ansible-playbook '
-            f'--ssh-common-args="{self._ssh_options(join=True)}" '
-            f'{self._ansible_verbosity()} '
-            f'{self._ansible_extra_args(extra_args)} -i {self._ssh_guest()},'
-            f' {playbook}',
-            cwd=self.parent.plan.worktree)
+            [f'ansible-playbook'] +
+            [f'{self._ansible_verbosity()}'] +
+            [f'{self._ansible_extra_args(extra_args)}'] +
+            ssh_common_args +
+            [f'-i'] +
+            [f'{self._ssh_guest()},'] +
+            [f'{playbook}'],
```

Comment:
Why are you using format strings even for variables which contain strings?
```suggestion
            [playbook],
```